### PR TITLE
Provide a method to access all of an enum's options

### PR DESCRIPTION
**Pull request by agrobbin**
Previously getting at custom configuration parameters was really nasty. This provides a more convenient access point.

### CODE REVIEW

File: lib/flexible_enum/name_configurator.rb [hunk 1]
```diff
@@ -35,8 +40,6 @@ def human_name_for(value)
       end
     end
 
-    private
-
     def element_info(value)
       elements.select{|e,c| c[:value] == value }.first
     end
```
Comment (jschneiderhan) [inline lib/flexible_enum/name_configurator.rb]: This method is now considered useful outside the NameConfigurator class?

**Comment by jschneiderhan** [comment]
Looks good to me.

File: lib/flexible_enum/name_configurator.rb (hunk 1 above)
Comment (agrobbin) [inline lib/flexible_enum/name_configurator.rb]: Oh, hmm, yeah maybe I should change that.

**Comment by MDaubs** [comment]
@agrobbin is there a real world example you can show of how this is used? I'm a little uneasy about returning custom configuration parameters without the name and human_name, which is generally how we identify options.

**Comment by agrobbin** [comment]
@MDaubs I have this model:

```ruby
class Rewards::Incentive < ActiveRecord::Base
  flexible_enum :incentive_type do
    gift_card 1, shippable: true
    medical_plan_discount 2, shippable: false
    health_day_off 3, shippable: false
    pedometer 4, shippable: false
  end
end
```

And would like to be able to get at the `:shippable` option like this:

```ruby
def shippable?
  incentive_type_options[:shippable]
end
```

File: lib/flexible_enum/name_configurator.rb (hunk 1 above)
Comment (jschneiderhan) [inline lib/flexible_enum/name_configurator.rb]: Really? I contributed a useful comment to a PR?

**Comment by MDaubs** [comment]
Oh I see, I overlooked that `enum_name_options` returns the options for the current value. I think this makes sense. One alternative we might consider is returning an object that responds to each configuration item ala `OpenStruct`. This would allow you to:
```ruby
delegate shippable?, to: :incentive_type_options`
```
Not sure if that's desirable or not. There's something nice about the simplicity of returning a `Hash`.

**Comment by agrobbin** [comment]
@MDaubs yeah I went back and forth on that, I kind of like the simplicity of the `Hash`, but can be persuaded to return an `OpenStruct` instead.

File: lib/flexible_enum/name_configurator.rb (hunk 1 above)
Comment (aprescott) [inline lib/flexible_enum/name_configurator.rb]: @jschneiderhan I won't stand for this self-deprecation! You're awesome.

File: lib/flexible_enum/name_configurator.rb [hunk 2]
```diff
@@ -3,6 +3,11 @@ class NameConfigurator < AbstractConfigurator
     def apply
       configurator = self
 
+      add_instance_method("#{attribute_name}_options") do
```
Comment (MDaubs) [inline lib/flexible_enum/name_configurator.rb]: `name_configurator.rb` contains logic for adding methods that expose the name variants of the current enum value. I think this new functionality isn't really about names. What do you think about either moving the options addition here to a new file or renaming `name_configurator.rb` to `identity_configurator.rb` (or similar) to better indicate that this file is about more than just names?

**Comment by MDaubs** [comment]
I tend to err on the side of `OpenStruct` since it often gives us flexibility to add logic later. Although in this case any logic we added to the object returned would potentially have method name collisions with enum names defined by applications. So maybe `Hash` is best. Eh. No strong opinion.

File: README.md [hunk 3]
```diff
@@ -213,13 +213,15 @@ class EmailEvent < ActiveRecord::Base
...
-  def process
-    class.event_types[event_type][:processor_class].new(self).process
-  end
 end
 ```
 
+Custom configuration parameters are available as an instance method on the object as well.
+```
+e = EmailEvent.new(event_type: 1)
+e.event_type_options # => { processor_class: RejectedProcessor, value: 1 }
```
Comment (MDaubs) [inline README.md]: We might consider renaming this to `event_type_details` or something else without the word "options". The word "option" is used elsewhere to mean an available enum value. Here, "options" means the configuration attributes of the current "option".

**Comment by aprescott** [comment]
@MDaubs I tend to dislike `OpenStruct` since returning `nil`s on arbitrary method calls feels about as bad as command line tools silently accepting all arguments. Moreso in library code.

File: lib/flexible_enum/name_configurator.rb (hunk 2 above)
Comment (agrobbin) [inline lib/flexible_enum/name_configurator.rb]: :+1: I like `identity_configurator.rb`.

**Comment by agrobbin** [comment]
@MDaubs @aprescott I'm going to stick with `Hash` for now.

**Comment by MDaubs** [comment]
Oh yes, good point @aprescott. I'm off the fence and in the `Hash` camp, too, now.

**Comment by aprescott** [comment]
Can we take this as an opportunity to create a `CHANGELOG.md` file or something similar?

**Pull request merged**